### ISSUE TITLE
Implicit Resolvers do not appear to work with PyYAML 5.1

### ISSUE THREAD

**Issue by cfeenstra67** [yaml/pyyaml#294]
I've noticed that with PyYAML 5.1 implicit resolvers do not seem to work at all. Using this example script:
```python
import re
import uuid
import yaml

regex = re.compile(r'^UUID\((.+)\)$')
yaml.add_implicit_resolver('!uuid', regex)

def convert_uuid(loader, node):
    value = loader.construct_scalar(node)
    str_value = regex.match(value).group(1)
    return uuid.UUID(str_value)

yaml.add_constructor('!uuid', convert_uuid)

print(yaml.load('''
    config:
        abc: UUID(6a02171e-6482-11e9-ab43-f2189845f1cc)
        def: abc
'''))
```
I confirmed that w/ 3.13 we correctly get the `uuid.UUID` object, while it is a string with 5.1 is it just the string `'UUID(6a02171e-6482-11e9-ab43-f2189845f1cc)'`. The way I found this was that it broke the environment variable interpolation that `elasticsearch-curator` provides. 

Sorry if this is a duplicate, I searched around and could not find an equivalent issue.

**Comment by cfeenstra67** [comment]
After looking around this appears to be the same issue as https://github.com/yaml/pyyaml/issues/266 , I confirmed it does work with `Loader` set. It does seem to be a bug, but I'll close this issue as it's covered there

**Comment by perlpunk** [comment]
@ingydotnet I suggest to reopen because `add_implicit_resolver` is not yet covered by the fixes.
(also `add_path_resolver` btw)

**Comment by perlpunk** [comment]
I'm reopening this for @ingydotnet

**Comment by perlpunk** [comment]
I created #305 to fix this

**Comment by perlpunk** [comment]
We just released 5.2b1 https://pypi.org/project/PyYAML/5.2b1/ which should fix this.

**Comment by perlpunk** [comment]
We released 5.2: https://pypi.org/project/PyYAML/5.2/

**Comment by perlpunk** [comment]
Closing. Please reopen if necessary. Thanks!
